### PR TITLE
Rbreuer add new token support to CLI

### PR DESCRIPTION
**Pull request by ranbreuer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ Imports a PBIX file into a
 Updates connection strings and/or credentials for an existing dataset
 
 `powerbi update-connection -c <collection> -w <workspaceId> -k <accessKey> -d <datasetId> -s [connectionString] -u [username] -p [password]`
-#### create-token
-Creates a Power BI app token
+#### create-embed-token
+Creates a Power BI embed app token
 
-`powerbi create-token -t <tokenType> -c <collection> -k <accessKey> -w <workspaceId> -r <reportId> -u [username] --roles [roles1,roles2,...]`
+`powerbi create-embed-token -c <collection> -k <accessKey> -w <workspaceId> -r <reportId> -u [username] --roles [roles1,roles2,...]`
```
Comment (ArinaHantsis) [inline README.md]: What about the v1 tokens? There used to be embed, management and dev token types? We're deprecating them staring with this version?

File: lib/cli-create-embed-token.ts [hunk 2]
```diff
@@ -16,13 +16,17 @@ export default function CliCreateEmbedToken() {
         .option('-k, --accessKey <accessKey>', 'The Power BI workspace collection access key')
         .option('-w, --workspace <workspaceId>', 'The Power BI workspace id')
         .option('-r, --report <reportId>', 'The Power BI report id')
+        .option('-d, --dataset <datasetId>', 'The Power BI dataset id')
+        .option('-s, --scopes [scopes]', 'The permission scopes allowed')
```
Comment (ArinaHantsis) [inline lib/cli-create-embed-token.ts]: Would it worth mentioning the available scope list in the readme file?

File: lib/cli-create-embed-token.ts [hunk 3]
```diff
@@ -16,13 +16,17 @@ export default function CliCreateEmbedToken() {
...
         .option('-u, --username [username]', 'The RLS username')
         .option('--roles [roles]', 'The RLS roles', list);
 
     program.on('--help', function () {
         console.log('  Examples:');
         console.log('');
         console.log('    $ powerbi create-embed-token -c <collection>, -w <workspace> -r <reportId> -k <accessKey> -u [username] --roles [roles1,roles2,...]');
+        console.log('    $ powerbi create-embed-token -c <collection>, -w <workspace> -d <datasetId> -k <accessKey>');
```
Comment (ArinaHantsis) [inline lib/cli-create-embed-token.ts]: I'd add the scopes the dataset embed token as it's mandatory for the create operation

File: README.md (hunk 1 above)
Comment (ranbreuer) [inline README.md]: they were never supported here, only bad naming

File: lib/cli-create-embed-token.ts (hunk 2 above)
Comment (ranbreuer) [inline lib/cli-create-embed-token.ts]: good idea

File: lib/cli-create-embed-token.ts [hunk 4]
```diff
@@ -39,10 +43,14 @@ export default function CliCreateEmbedToken() {
...
             }
 
-            if (!(settings.collection && settings.workspace && settings.report)) {
-                return cli.error('collection, workspace and report params are required');
+            if (!(settings.collection && settings.workspace)) {
+                return cli.error('collection and workspace params are required');
             }
-            token = powerbi.PowerBIToken.createReportEmbedToken(settings.collection, settings.workspace, settings.report, settings.username, settings.roles);
+
+            if (!(settings.report || settings.dataset)) {
```
Comment (yuryb2) [inline lib/cli-create-embed-token.ts]: What if both are specified?

**Pull request merged**